### ISSUE TITLE
recursive dependency management?

### ISSUE THREAD

**Issue by bendichter** [tomdoel/depmat#1]
Is there was any interest here in making this a recursive dependency management? As an example of how this might work, there could be a file like "matdep.txt" in the parent dir of the project, and when a repo is added as a dependency, all of the repos in this file are also added. That way, you could easily install new packages that depend on other libraries (that themselves depend on other libraries, etc.)

**Comment by tomdoel** [comment]
Interesting idea. What happens if two repos depend on different branches of the same repo? Probably the one defined highest in the hierarchy would take precedence, and a warning would be displayed.

**Comment by bendichter** [comment]
Good point, though I don't think it is specific to this feature. You could still have this problem (though this would be less likely) if two different branches of the same repo were defined in a single RepoList. 

I like your solution. In my opinion, relying on different branches of the same repo is a mess and warrants a warning. I know python deals with this type of problem similarly, and has the capability to make multiple environments if you really need both branches.
